src/RawPOSIX/src/safeposix/dispatcher.rs

```diff
@@ -774,12 +775,52 @@ pub fn lind_syscall_api(
 
         SHMAT_SYSCALL => {
             let shmid = arg1 as i32;
-            let cage = interface::cagetable_getref(cageid);
-            // Convert virtual address to physical address
-            let shmaddr = translate_vmmap_addr(&cage, arg2).unwrap() as *mut u8;
+            let shmaddr = arg2 as *mut u8; // No need to translate - shmat_handler will handle address translation
```

Comment:
Its strange to me that you seem to implement shmat directly in the dispatcher while shmdt from the inner syscall. We should be consistent in coding style here and I'd defer to matching what we did with the mmap syscalls for now.